src/query/storages/result_cache/src/write/sink.rs

```diff
@@ -51,18 +59,32 @@ impl AsyncMpscSink for WriteResultCacheSink {
     #[async_trait::unboxed_simple]
     #[async_backtrace::framed]
     async fn consume(&mut self, block: DataBlock) -> Result<bool> {
+        if self.terminated {
+            return Ok(true);
+        }
+
+        if !self.consumed_one_block {
+            if self.cache_writer.not_over_time(&self.create_time) {
+                // Skip the cache writing if the query execution time is less than the min_execute_secs.
```

Comment:
It may not indicate the query is finished.

Reply:
It indicates first query block reach time.